pisces/data_sets.py

```diff
@@ -327,6 +334,440 @@ def find_data_sets(cls, root: str | Path) -> Dict[str, 'DataSetObject']:
+            for i in range(new_data.shape[1])
+        }})
+
+    if smooth:
+        acc_resampled = apply_gausian_filter(acc_resampled, overwrite=True)
+
+    return acc_resampled
+
+# %% ../nbs/01_data_sets.ipynb 18
+class DataProcessor:
+    def __init__(self, 
+                 data_set: DataSetObject,
+                 model_input: ModelInput,
+                 output_feature: str='psg',
+                 output_type: ModelOutputType=ModelOutputType.WAKE_LIGHT_DEEP_REM,
+                 psg_type: PSGType=PSGType.HAS_N4,
+                 ):
+        self.data_set = data_set
+        self.input_features = model_input.input_features
+        self.output_feature = output_feature
+        self.output_type = output_type
+        self.psg_type = psg_type
+        self.model_input = model_input
+
+        if isinstance(model_input, ModelInput1D):
+            self.input_window_time = model_input.input_window_time
+            self.input_sampling_hz = model_input.input_sampling_hz
+            self.input_window_samples = model_input.input_window_samples
+            self.model_input_dimension = model_input.model_input_dimension
+        elif isinstance(model_input, ModelInputSpectrogram):
+            self.input_sampling_hz = model_input.input_sampling_hz
+            self.spectrogram_preprocessing_config = model_input.spectrogram_preprocessing_config
+
+    def get_labels(self, id: str, start: int, end: int,
+                   output_feature: str) -> pl.DataFrame | None:
+        data = self.data_set.get_feature_data(output_feature, id)
+        data = data.filter(data[:, 0] >= start)
+        data = data.filter(data[:, 0] <= end)
+
+        # Mask PSG data based on accelerometer data if present
+        if "accelerometer" in self.input_features:
+            accelerometer = self.data_set.get_feature_data('accelerometer', id)
+            data = mask_psg_from_accel(data, accelerometer, 
+                                       accel_sample_rate=self.input_sampling_hz)
+
+        if self.output_feature == 'psg':
+            if self.output_type == ModelOutputType.SLEEP_WAKE:
+                y = psg_to_sleep_wake(data)
+            elif self.output_type == ModelOutputType.WAKE_LIGHT_DEEP_REM:
+                N4 = self.psg_type == PSGType.HAS_N4
+                y = psg_to_WLDM(data, N4)
+            else:
+                raise ValueError(f"Output type {self.output_type} not supported")
+        else:
+            raise ValueError(f"Output feature {output_feature} not supported")
+
+        labels = pl.DataFrame({
+            'time': data[:, 0],
+            'label': y,
+        })
+
+        return labels
+
+    def get_1D_X_for_feature(self, interpolation_timestamps: np.ndarray, 
+                             epoch_times: np.ndarray, feature_times: np.ndarray, 
+                             feature_values: np.ndarray) -> np.ndarray:
+            interpolation = np.interp(interpolation_timestamps, feature_times, feature_values)
```

Comment:
There is an issue here when `feature_values` is not 1d. You must interpolate each axis separately, as far as I know.

For example:

```python
import numpy as np

t = np.linspace(0, 2*np.pi, 100)
x = np.zeros((len(t), 2))
x[:, 0] = np.sin(t)
x[:, 1] = np.cos(t)

t2 = np.linspace(0, 2*np.pi, 50)
x2 = np.interp(t2, t, x)
```
outputs
```markdown
{
	"name": "ValueError",
	"message": "object too deep for desired array",
	"stack": "---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[3], line 8
      4 x[:, 1] = np.cos(t)
      6 t2 = np.linspace(0, 2*np.pi, 50)
----> 8 x2 = np.interp(t2, t, x)

File ~/Engineering/Work/pisces/.pisces/lib/python3.11/site-packages/numpy/lib/function_base.py:1599, in interp(x, xp, fp, left, right, period)
   1596     xp = np.concatenate((xp[-1:]-period, xp, xp[0:1]+period))
   1597     fp = np.concatenate((fp[-1:], fp, fp[0:1]))
-> 1599 return interp_func(x, xp, fp, left, right)

ValueError: object too deep for desired array"
}
```

Reply:
You're right, I updated the code on `get_1D_X_y` to go through all the columns in a feature and feed them one by one to this function. Do you think that's a good idea?